New resource: Dedicated Host (#5417)
terraform-providers/terraform-provider-azurerm: website/docs/r/dedicated_host.html.markdown

---
subcategory: "Compute"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_dedicated_host"
description: |-

>   Manage an Dedicated Host.

Reviewer: ```suggestion
  Manage a Dedicated Host within a Dedicated Host Group.
```

---

# azurerm_dedicated_host

> Manage an Dedicated Host.

Reviewer: ```suggestion
Manage a Dedicated Host within a Dedicated Host Group.
```

## Example Usage

```hcl
resource "azurerm_resource_group" "example" {

>   name     = "example-rg"

Reviewer: ```suggestion
  name     = "example-resources"
```

>   location = "West US"

Reviewer: we use West Europe for all examples

```suggestion
  location = "West Europe"
```

  host_group_name       = azurerm_dedicated_host_group.example.name
  sku_name              = "DSv3-Type1"
  platform_fault_domain = 1
}
```

## Argument Reference

The following arguments are supported:

* `name` - (Required) Specify the name of the Dedicated Host. Changing this forces a new resource to be created.

* `resource_group_name` - (Required) The name of the resource group in which to create the Dedicated Host. Changing this forces a new resource to be created.

* `location` - (Required) Specify the supported Azure location where the resource exists. Changing this forces a new resource to be created.

* `host_group_name` - (Required) Specify the name of the Dedicated Host Group in which to create the Dedicated Host. Changing this forces a new resource to be created.

* `sku_name` - (Required) Specify the sku name of the Dedicated Host. Possible values are `DSv3-Type1`, `ESv3-Type1`, `FSv2-Type2`. Changing this forces a new resource to be created.

* `platform_fault_domain` - (Required) Specify the fault domain of the Dedicated Host Group in which to create the Dedicated Host. Changing this forces a new resource to be created.

> * `auto_replace_on_failure` - (Optional) Specifies whether the Dedicated Host should be replaced automatically in case of a failure. The value is defaulted to `true` when not provided.

Reviewer: to make this consistent with other resources can we make this:

```suggestion
* `auto_replace_on_failure` - (Optional) Specifies whether the Dedicated Host should be replaced automatically in case of a failure. Defaults to `true`.
```
